### PR TITLE
Blood Cult Conversion Fixes

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
fix: Twisted Construction will now consume ALL available plasteel in a stack.
fix: Runes will no longer count the original invoker more than once. 
/:cl:

### CODE REVIEW

**Comment by Robustin** [comment]
Converting plasteel 1 unit at a time was not fun...

Cheesing rune requirements isn't particularly fun either.

**Comment by ExcessiveUseOfCobblestone** [comment]
you know you could just remove user from invokers in that pick so you don't have to check if the person is the user each time

File: code/modules/antagonists/cult/blood_magic.dm [hunk 1]
```diff
@@ -546,12 +546,13 @@
...
-			uses -= quantity
-			new /obj/item/stack/sheet/runed_metal(T,quantity)
-			candidate.use(quantity)
-			to_chat(user, "<span class='warning'>A dark cloud eminates from you hand and swirls around the plasteel, transforming it into runed metal!</span>")
-			SEND_SOUND(user, sound('sound/effects/magic.ogg',0,1,25))
+			var/quantity = candidate.amount
+			if(candidate.use(quantity))
+				uses --
+				new /obj/item/stack/sheet/runed_metal(T,quantity)
+				candidate.use(quantity)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm]: why are you calling this twice

**Comment by Robustin** [comment]
That was my first instinct too but I recall there being a reason for wanting the user to remain in the invoker list, like I think some runes will check invoker[1] - this is the best way to ensure I don't break anything.

File: code/modules/antagonists/cult/blood_magic.dm (hunk 1 above)
Comment (Robustin) [reply code/modules/antagonists/cult/blood_magic.dm]: ah shit, made that change to an if(use) after testing - good catch

**Comment by ExcessiveUseOfCobblestone** [comment]
it's not removing them from the invoker list, it's removing them from the pick options

**Comment by Robustin** [comment]
You mean like:

var/C = pick_n_take(invokers - user)?

**Comment by ExcessiveUseOfCobblestone** [comment]
y-y-y-yes

**Pull request merged**